marlowe-playground-client/src/Simulation.purs

```diff
@@ -400,616 +276,429 @@ marloweActionInput isEnabled f current =
+
+  hasRuntimeWarnings = state ^. (_marloweState <<< _Head <<< _transactionWarnings <<< to Array.null <<< to not)
+
+  hasRuntimeError = state ^. (_marloweState <<< _Head <<< _transactionError <<< to isJust)
+
   contractMaxTime Nothing = "Closed"
 
   contractMaxTime (Just contract) = let t = maxTime contract in if t == zero then "Closed" else show t
 
-statePane :: forall p. FrontendState -> HTML p HAction
-statePane state =
-  div
-    [ class_ $ ClassName "col"
-    ]
-    [ stateTable state
-    ]
-
-stateTable :: forall p. FrontendState -> HTML p HAction
-stateTable state =
-  div_
-    [ card_
-        [ cardBody_
-            [ h3_
-                [ text "Accounts"
-                ]
-            , div
-                [ classes [ ClassName "state-row", ClassName "full-width-card-5" ] ]
-                [ if (Map.size accounts == 0) then
-                    text "There are no accounts in the state"
-                  else
-                    renderAccounts accounts
-                ]
-            , h3_
-                [ text "Choices"
-                ]
-            , div
-                [ classes [ ClassName "state-row", ClassName "full-width-card" ] ]
-                [ if (Map.size choices == 0) then
-                    text "No choices have been recorded"
-                  else
-                    renderChoices choices
-                ]
-            , h3_
-                [ text "Payments"
-                ]
-            , div
-                [ classes [ ClassName "state-row", ClassName "full-width-card-4" ] ]
-                [ if (Array.length payments == 0) then
-                    text "No payments have been recorded"
-                  else
-                    renderPayments payments
-                ]
-            , h3_
-                [ text "Let bindings"
-                ]
-            , div
-                [ classes [ ClassName "state-last-row", ClassName "full-width-card" ] ]
-                [ if (Map.size bindings == 0) then
-                    text "No values have been bound"
-                  else
-                    renderBindings bindings
-                ]
-            ]
-        ]
+panelContents :: forall p. FrontendState -> SimulationBottomPanelView -> HTML p HAction
+panelContents state CurrentStateView =
+  div [ class_ Classes.panelContents ]
+    [ div [ classes [ rTable, rTable6cols, ClassName "panel-table" ] ]
+        ( warningsRow <> errorRow
+            <> tableRow "Accounts" "No accounts have been used" "Account ID" "Participant" "Currency Symbol" "Token Name" "Money"
+                accountsData
+            <> tableRow "Choices" "No Choices have been made" "Choice ID" "Participant" "Chosen Value" "" ""
+                choicesData
+            <> tableRow "Payments" "No payments have been recorded" "Party" "Currency Symbol" "Token Name" "Money" "" paymentsData
+            <> tableRow "Let Bindings" "No values have been bound" "Identifier" "Value" "" "" "" bindingsData
+        )
```

Comment:
The padding with empty strings should be done by a function that should probably have a type like `[([String], AccountDatastype)]`

Reply:
again, the list problem applies, I want a fixed number of arguments to ensure layout is correct. However I have changed `""` to `mempty` because I feel like it gives a bit more information about what we are doing here. Not much but a bit. If you can think of something better we can do I'd be happy to change. (FYI this is in `src/Simulation/BottomPanel.purs` now